tethys_apps/base/app_base.py

```diff
@@ -924,6 +925,53 @@ def get_custom_setting(cls, name):
         except ObjectDoesNotExist:
             raise TethysAppSettingDoesNotExist('CustomTethysAppSetting', name, cls.name)
 
+    @classmethod
+    def set_custom_setting(cls, name, value):
+        """
+        Assign the value of a CustomSetting for the app.
+
+        Args:
+            name(str): The name of the CustomSetting as defined in the app.py.
+            value(str/int/float/boolean/uuid.UUID): the value of the customSetting.
+
+        Returns:
+            variable: Value of the CustomSetting or None if no value assigned.
+
+        **Example:**
+
+        ::
+
+            from my_first_app.app import MyFirstApp as app
+
+            max_count = app.set_custom_setting('max_count', 5)
+
+        """
+        from tethys_apps.models import TethysApp
+        db_app = TethysApp.objects.get(package=cls.package)
+        custom_settings = db_app.custom_settings
+        try:
+            custom_setting = custom_settings.get(name=name)
+        except ObjectDoesNotExist:
+            raise TethysAppSettingDoesNotExist('CustomTethysAppSetting', name, cls.name)
+
+        type_matches = False
+        if custom_setting.type == 'STRING':
+            type_matches = isinstance(value, str)
+        elif custom_setting.type == 'INTEGER':
+            type_matches = isinstance(value, int)
+        elif custom_setting.type == 'FLOAT':
+            type_matches = isinstance(value, float)
+        elif custom_setting.type == 'BOOLEAN':
+            type_matches = str(value).lower() in ['true', 'false', 'yes', 'no', 't', 'f', 'y', 'n', '1', '0']
+        elif custom_setting.type == 'UUID':
+            type_matches = isinstance(value, uuid.UUID)
```

Comment:
Technically they could provide a string version of a UUID that I would think we count as valid:

```
>>> str(uuid.uuid4())
'f50ec0b7-f960-400d-91f0-c42a6d44e3d0'
```

In addition to checking if it is an instance of a `uuid.UUID` I suggest you try to instantiate a `uuid.UUID`  from the given value.

Reply:
You are right. I should have caught that since I did it something similar with the boolean type.